src/main/grammars/RustParser.bnf

```diff
@@ -504,26 +504,32 @@ upper Function ::= async? const? unsafe? ExternAbi?
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
-Pat ::= PatWild
-      | PatRest
-      | PatRef
-      | PatTup
-      | PatSlice
-      | PatIdent
-      | PatMacro
-      | PatStruct
-      | PatTupleStruct
-      | (PatConst !('..' | '...' | '..='))
-      | PatRange
-      | PatBox {
+private TopPat ::= &'|' OrPat | !'!' Pat
```

Comment:
```suggestion
private TopPat ::= &'|' OrPat | !'!' Pat { name = "pattern" }
```
Fixes chanes in `partial/exprs.txt` parsing test

Reply:
Also, why `!'!'` is needed? Let's add a test in `RsCompleteParsingTestCase` for this case

Reply:
> Also, why !'!' is needed?

:man_facepalming:
It should be `!'|'`. But it doesn't need anyway - just artifact of development. Thanks!